src/NuGet.Clients/NuGet.VisualStudio.Internal.Contracts/ContextInfos/PackageSearchMetadataContextInfo.cs

```diff
@@ -73,7 +73,7 @@ public static PackageSearchMetadataContextInfo Create(IPackageSearchMetadata pac
                 IsListed = packageSearchMetadata.IsListed,
                 DependencySets = packageSearchMetadata.DependencySets?.ToList(),
                 DownloadCount = packageSearchMetadata.DownloadCount,
-                Vulnerabilities = packageSearchMetadata.Vulnerabilities?.Select(vulnerability => new PackageVulnerabilityMetadataContextInfo(vulnerability.AdvisoryUrl, vulnerability.Severity)).ToArray(),
+                Vulnerabilities = packageSearchMetadata.Vulnerabilities?.Select(vulnerability => new PackageVulnerabilityMetadataContextInfo(vulnerability.AdvisoryUrl, vulnerability.Severity)).OrderByDescending(v => v.Severity).ToArray(),
```

Comment:
Please check if we can sort vulnerabilities after receiving from the protocol. The reason is, I think all the http responses are cached for certain period of time. If the vulnerabilities are sorted before caching then when there is a cache hit the data will already be sorted there by improving the performance.

Reply:
@kartheekp-ms can you point me to where that is? Thanks!

Reply:
Hey @jebriede sorry if my comment was not actionable. The closest location in the source code where `vulnerabilities` can be sorted is here. https://github.com/NuGet/NuGet.Client/blob/3ec538cb909cf5b9b0f663ef6f7a6891406d21aa/src/NuGet.Core/NuGet.Protocol/Model/PackageSearchMetadataBuilder.cs#L110

Please note, I didn't debug the exact flow of the code. Please verify because there could be a gap in my understanding on the respective code path.

Reply:
Kartheek raises a good point, but after a brief think through it, I believe a maximum of 4 `int`s being sorted is probably low impact. I'm also not sure which state we would want the cache to be in, or if we could rely on it not becoming stale.

Reply:
Thanks for your feedback @kartheekp-ms. I agree with the points raised by @donnie-msft in that the sorting is low impact and sorting at a lower layer would affect other areas including the cache while sorting at this layer will affect only how it's displayed.